### PR TITLE
Fix closing settings overlay with key binding panel opened can cause unwanted behavior

### PR DESCRIPTION
**Pull request by EVAST9919**

### CODE REVIEW

File: osu.Game/Overlays/SettingsOverlay.cs [hunk 1]
```diff
@@ -82,5 +82,18 @@ private void load()
...
+        {
+            base.PopOut();
+
+            foreach (var s in subPanels)
+            {
+                if (s.State.Value == Visibility.Visible)
+                {
+                    s.Hide();
+                }
+            }
```
Comment (iiSaLMaN) [inline osu.Game/Overlays/SettingsOverlay.cs]: can be simplified into
```csharp
foreach (var s in subPanels)
    if (s.State.Value == Visibility.Visible)
        s.Hide();
```

**Comment by Joehuu** [comment]
PR #4857 is fixing the same problem.

**Comment by EVAST9919** [comment]
Depends on user's expectations. The question is do we want to close all the settings subpanels when closing the settings or just fix that like in https://github.com/ppy/osu/pull/4857

**Comment by peppy** [comment]
I think it is more user-friendly to remember the last location viewed, personally.

**Comment by EVAST9919** [comment]
as being said

**Pull request closed without merging**